src/core/sock/sockinfo.cpp

```diff
@@ -425,6 +427,19 @@ int sockinfo::setsockopt(int __level, int __optname, const void *__optval, sockl
                           "optval == NULL");
             }
             break;
+
+        case SO_REUSEADDR:
+            m_reuseaddr = *(bool *)__optval;
+            si_logdbg("SOL_SOCKET, %s=%s", setsockopt_so_opt_to_str(__optname),
+                      (*(bool *)__optval ? "true" : "false"));
```

Comment:
The boolean value is already stored in `m_reuseaddr`. Using it will be more readable.